helix-term/src/ui/picker.rs

```diff
@@ -113,17 +146,35 @@ impl<T: 'static> Component for FilePicker<T> {
             horizontal: 1,
         };
         let inner = inner.inner(&margin);
-
         block.render(preview_area, surface);
 
-        if let Some((doc, line)) = self.current_file(cx.editor).and_then(|(path, range)| {
-            cx.editor
-                .document_by_path(&path)
-                .or_else(|| self.preview_cache.get(&path))
-                .zip(Some(range))
-        }) {
+        if let Some((path, range)) = self.current_file(cx.editor) {
+            let cache = self.preview_cache.get(&path);
+            let doc = match cache {
+                Some(Preview::Binary | Preview::LargeFile | Preview::NotFound) => {
+                    surface.set_stringn(
+                        inner.x,
+                        inner.y,
+                        match cache {
+                            Some(Preview::Binary) => "<Binary file>",
```

Comment:
Centering the placeholder texts would be nice to have.

Reply:
Are there any implementation that I can reference to center the placeholder text?
I'm stacked.

Reply:
I think you need to calculate it by hand (`let y = inner.y + inner.width / 2`) etc. we can do it after this PR

Reply:
Ok, I skip this in this PR.